VocaDbModel/Database/Queries/SongListQueries.cs

```diff
@@ -301,7 +303,7 @@ public class SongListQueries : QueriesBase<ISongListRepository, SongList> {
 					list = ctx.Load(contract.Id);
 					var diff = new SongListDiff();
 
-					EntryPermissionManager.VerifyEdit(PermissionContext, list);
+					EntryPermissionManager.VerifyEdit(PermissionContext, (ISongList)list);

```

Comment:
Why is this cast needed?

Reply:
Because the call is ambiguous between `EntryPermissionManager.CanEdit(IUserPermissionContext, ISongList)` and `EntryPermissionManager.CanEdit(IUserPermissionContext, IEntryWithStatus)`. Actually, SongList is enough for EntryPermissionManager.VerifyEdit.

Reply:
Then I recommend that you should add CanEdit overloads for the concrete types so that there is no ambiguity. Or rename it to CanEditSongList. I would prefer getting rid of the casts if possible.